packages/eslint-plugin/tests/rules/array-type.test.ts

```diff
@@ -10,175 +10,201 @@ ruleTester.run('array-type', rule, {
       code: `function bazFunction(baz: Arr<ArrayClass<String>>) {
     return baz.map(e => e.baz);
 }`,
-      options: ['array'],
+      options: [{ default: 'array' }],
     },
     {
       code: `let barVar: ((c: number) => number)[];`,
-      options: ['array'],
+      options: [{ default: 'array' }],
     },
     {
       code: `type barUnion = (string|number|boolean)[];`,
-      options: ['array'],
+      options: [{ default: 'array' }],
     },
     {
       code: `type barIntersection = (string & number)[];`,
-      options: ['array'],
+      options: [{ default: 'array' }],
     },
     {
       code: `interface FooInterface {
     '.bar': {baz: string[];};
 }`,
-      options: ['array'],
+      options: [{ default: 'array' }],
     },
     {
       // https://github.com/typescript-eslint/typescript-eslint/issues/172
       code: 'type Unwrap<T> = T extends (infer E)[] ? E : T',
-      options: ['array'],
+      options: [{ default: 'array' }],
     },
     {
       code: `let z: Array = [3, "4"];`,
-      options: ['generic'],
+      options: [{ default: 'generic' }],
     },
     {
       code: `let xx: Array<Array<number>> = [[1, 2], [3]];`,
-      options: ['generic'],
+      options: [{ default: 'generic' }],
     },
     {
       code: `type Arr<T> = Array<T>;`,
-      options: ['generic'],
+      options: [{ default: 'generic' }],
     },
     {
       code: `function fooFunction(foo: Array<ArrayClass<string>>) {
     return foo.map(e => e.foo);
 }`,
-      options: ['generic'],
+      options: [{ default: 'generic' }],
     },
     {
       code: `function bazFunction(baz: Arr<ArrayClass<String>>) {
     return baz.map(e => e.baz);
 }`,
-      options: ['generic'],
+      options: [{ default: 'generic' }],
     },
     {
       code: `let fooVar: Array<(c: number) => number>;`,
-      options: ['generic'],
+      options: [{ default: 'generic' }],
     },
     {
       code: `type fooUnion = Array<string|number|boolean>;`,
-      options: ['generic'],
+      options: [{ default: 'generic' }],
     },
     {
       code: `type fooIntersection = Array<string & number>;`,
-      options: ['generic'],
+      options: [{ default: 'generic' }],
     },
     {
       // https://github.com/typescript-eslint/typescript-eslint/issues/172
       code: 'type Unwrap<T> = T extends Array<infer E> ? E : T',
-      options: ['generic'],
+      options: [{ default: 'generic' }],
+    },
+
+    // readonly
+    {
+      code: 'let a: string[] = []',
+      options: [{ default: 'array', readonly: 'generic' }],
+    },
+    {
+      code: 'let a: ReadonlyArray<number> = []',
+      options: [{ default: 'array', readonly: 'generic' }],
+    },
+    {
+      code: 'let a: ReadonlyArray<number[]> = [[]]',
+      options: [{ default: 'array', readonly: 'generic' }],
+    },
+    {
+      code: `let a: Array<string> = []`,
+      options: [{ default: 'generic', readonly: 'array' }],
+    },
+    {
+      code: `let a: readonly Array<number> = []`,
```

Comment:
this is not valid typescript
```suggestion
      code: `let a: readonly number[] = []`,
```

Reply:
@bradzacher Oops! Will fix. Thanks. :+1: